packages/liveblocks-client/src/room.ts

```diff
@@ -561,7 +563,8 @@ export function makeStateMachine<TPresence extends JsonObject>(
       },
     };
 
-    const createdNodeIds = new Set<string>();
+    const createdNodeIds = new Set<string | undefined>();
+    //                                      ^^^^^^^^^ NOTE: Bug? Unintended?
```

Comment:
Here's one case where the unit test suite pointed out a case where we annotate that this Set only ever contains strings, but in reality it also relies on `.add(undefined)` and `.has(undefined)` checks. This may need some investigation.

Reply:
Parent can be undefined if it's the root. Thank you for finding this!